test/e2e/apimachinery/aggregator.go

```diff
@@ -291,6 +291,12 @@ func TestSampleAPIServer(f *framework.Framework, image string) {
 	})
 	framework.ExpectNoError(err, "creating role binding %s:sample-apiserver to access configMap", namespace)
 
+	// Wait for the extension apiserver to be up and healthy
+	// kubectl get deployments -n <aggregated-api-namespace> && status == Running
+	// NOTE: aggregated apis should generally be set up in there own namespace (<aggregated-api-namespace>). As the test framework
+	// is setting up a new namespace, we are just using that.
+	err = framework.WaitForDeploymentComplete(client, deployment)
```

Comment:
lgtm. Not sure it's the source for the flakes, but it could be.

Reply:
Do we need add a `framework.ExpectNoError` call to ensure `err` is nil?

Reply:
yes, added